semantic-analysis/src/Analysis/Reference.hs

```diff
@@ -0,0 +1,25 @@
+module Analysis.Reference
+( -- * Reference
+  Reference(..)
+  -- * Constructors
+, fromPath
+) where
+
+import Source.Span
+import System.Path as Path
+import System.Path.PartClass as Path.PartClass
+
+-- Reference
+
+data Reference = Reference
+  { refPath :: Path.AbsRelFile
+  , refSpan :: Span
+  }
+  deriving (Eq, Ord, Show)
```

Comment:
Two of the fields from `File`, bundled up together so we can use them, well, all over the place. They’re already pretty widely used and I’ll add them to closures eventually too.

Reply:
Incidentally, I’m not convinced that this package is really the best home for this, `File`, and similar data. I’ll figure out a better one later.